Adds NQueensBoard to the Environment [WIP]

samagra14 opened the pull request:
This PR aims at adding NQueensBoard to the set Environments available. This PR accomplishes the following tasks : 
- [x] GridLocation2D (To represent locations on a grid or a board)
- [x] NQueensBoard
- [ ] Add explanatory javadoc comments
- [ ] Add tests for the NQueens Environment
This environment can be used to demonstrate as well as test a large number of Search algorithms.

samagra14 committed:
Adds NQueensBoard to the Environment

RuedigerLunde commented:
The NQueensBoard class is a slightly changed copy of the corresponding class in the 3e branch. Why have comments and authors been removed? What is the improvement here?

samagra14 commented:
@RuedigerLunde
I had added the label [WIP] (Work in Progress) with the PR so that others can know I was working on this. But, somehow I forgot to mention it explicitly. And due to this miscommunication, it was merged before I could complete the documentation and testing of the environment. I have finished the relevant documentation and testing in a subsequent PR #392 .
There were few errors in the implementation of the nQueens environment in the AIMA3e branch. For instance, if there were three queens in a straight row or column, the environment counts them as three attacking pairs, however, the two queens at the extremes should not be counted as an attacking pair. Besides, I was planning on writing tests for the IterativeDeepeningSearch algorithm using the nQueens environment. Hence I implemented this environment.
Again, I am sorry for the miscommunication. You can suggest the relevant changes at #392 .

RuedigerLunde commented:
@Samagra Sharma
By dropping author information you steal intellectual property of others. Please stop doing that. You can copy code from the 3e branch, but in this case, all changes to it should be motivated by improvement only. This is still not the case in PR #392.
